src/components/ecology.jsx

```diff
@@ -3,42 +3,44 @@ import API from "./api";
 import Overview from "./overview";
 
 export default class Ecology extends React.Component {
-  renderAPI(source) {
```

Comment:
Right, that makes sense now. I've refactored that method in the latest commit. I do agree its better to put the logic in the method.